01.general/02.how-to-create-your-own-ad-filters/docs.en.md

```diff
@@ -427,6 +427,24 @@ There is a set of modifiers, which can be used to limit the rule's application a
 * `||example.org^$script,stylesheet` — corresponds to all the scripts and styles from `example.org`.
 * `||example.org^$~image,~script,~stylesheet` — corresponds to all requests to `example.org` except for the images, scripts and styles.
 
+<a id="document-modifier"></a>
+##### **`document`**
+
+The rule corresponds to HTML document requests, it applies to main frames only (i.e. HTML documents).
+
+In the case of blocking rules, `$document` also disables "main frame bypass" logic and blocks requests _strictly_ with a blocking page.
+
+In case of allowlist it completely disables blocking on corresponding pages. It is equal to simultaneous use of `elemhide`, `content`, `urlblock`, `jsinject` and `extension`.
+
+> **Compatibility with different versions of AdGuard.** Blocking request type logic now only supported by dev-build of AdGuard.
+
+###### `document` example
+
+* `@@||example.com^$document` — completely disables filtering on all pages at `example.com` and all subdomains.
+* `@@||example.com^$document,~extension` — completely disables blocking on any pages at `example.com` and all subdomains, but continues to run userscripts there.
+
```

Comment:
Please add examples when `$document` is used alongside `$redirect`, `$removeparam` or `$replace`.

Reply:
updated